### PR TITLE
Migrate from setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Using `ini2toml` package, I have converted `setup.cfg` to `pyproject.toml` entries.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -586,3 +797,56 @@ module = [
...
+]
+
+[tool.coverage.paths]
+source = [
+    "napari/",
+    "D:\\a\\napari\\napari\\napari",
+    "/home/runner/work/napari/napari/napari",
+    "/Users/runner/work/napari/napari/napari",
+]
+builtins = "napari_builtins/\nD:\\a\\napari\\napari\\napari_builtins\n/home/runner/work/napari/napari/napari_builtins\n/Users/runner/work/napari/napari/napari_builtins"
```
Comment (psobolewskiPhD) [inline pyproject.toml]:
```suggestion
builtins = [
    "napari_builtins/",
    "D:\\a\\napari\\napari\\napari_builtins",
    "/home/runner/work/napari/napari/napari_builtins",
    "/Users/runner/work/napari/napari/napari_builtins",
]
```

File: pyproject.toml [hunk 2]
```diff
@@ -586,3 +797,56 @@ module = [
...
+
+[tool.importlinter]
```
Comment (psobolewskiPhD) [inline pyproject.toml:821]: Based on the docs: https://import-linter.readthedocs.io/en/stable/usage.html#top-level-configuration
The quotes and \n should be removed.

**Comment by Czaki** [comment]
Next to migrate from `setup.cfg` to `pyproject.toml` I try to use `uv` for minimum testing, but currently `lowest` mode tries to apply the first existing release on PyPI for unbound dependencies, so cannot be applied in our case. But maybe they will add some heurestic for this. 
https://github.com/astral-sh/uv/issues/1718#issuecomment-1953791607

Currently, `lowest-direct` has the same result as `tox-min-req` plugin. So I think it should be removed at this moment.

**Review by jni — COMMENTED** [review]
This has consequences for napari/packaging, right? We should probably coordinate with that repo to make sure that they can keep following dependency changes here? CC @jaimergp

File: pyproject.toml (hunk 2 above)
Comment (jni) [reply pyproject.toml:821]: @Czaki what's the status with this comment?

**Comment by jaimergp** [comment]
Thanks for the tag! Yes, we tracked `setup.cfg` but we can just track `pyproject.toml` now :) Maybe even with a parser for the right table.

File: pyproject.toml (hunk 2 above)
Comment (Czaki) [reply pyproject.toml:821]: Debugged, and fixed

**Review by psobolewskiPhD — APPROVED** [review]
Looks good now.
tested locally:
`  Building editable for napari (pyproject.toml) ... done`
everything gucci

**Review by melonora — APPROVED** [review]
Also tested locally, LGTM

File: pyproject.toml [hunk 3]
```diff
@@ -1,11 +1,222 @@
...
+    "Operating System :: Unix",
+    "Operating System :: MacOS",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "appdirs>=1.4.4",
+    "app-model>=0.2.2,<0.3.0",
+    "cachey>=0.2.1",
+    "certifi>=2018.1.18",
+    "dask[array]>=2.15.0,!=2.28.0",
```
Comment (GenevieveBuckley) [inline pyproject.toml]:
```suggestion
    "dask[array]>=2021.10.0",
```

Comment (GenevieveBuckley) [reply pyproject.toml]: Addresses https://github.com/napari/napari/pull/6698 (I've also opened https://github.com/napari/napari/pull/6698, wasn't sure what the intended time frame for this PR is)

**Review by GenevieveBuckley — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (psobolewskiPhD) [reply pyproject.toml]: I think it's better to change the min in it's own PR? Gives future us clearer history. 

Plus the bump will require rebuilding min_req constraints.

Comment (Czaki) [reply pyproject.toml]: There is no `min_req` constraints, there is `tox-min-req` plugin that solves this problem and does not require bumping constraints file. 

I will merge #6698 and then merge this and fix.

**Review by GenevieveBuckley — APPROVED** [review]

**Comment by jni** [comment]
@jaimergp please make sure to update the sync script in napari/packaging! 🙏

**Pull request merged**